### PR TITLE
Observability task: add metadata buried points at the provider responder interface level

### PR DESCRIPTION
**Pull request by dengWuuu**
## What is the purpose of the change
related issue
https://github.com/apache/dubbo/issues/11764

### CODE REVIEW

**Comment by songxiaosheng** [comment]
@mytang0 time window test is not stable

**Review by songxiaosheng — APPROVED** [review]
LGTM

**Review by AlbumenJ — APPROVED** [review]

**Comment by AlbumenJ** [comment]
Please fix conflicts

**Comment by mytang0** [comment]
> @mytang0 time window test is not stable

Already fixed, #11848

**Comment by dengWuuu** [comment]
> Please fix conflicts

done

File: dubbo-metrics/dubbo-metrics-metadata/src/test/java/org/apache/dubbo/metrics/metadata/MetadataStatCompositeTest.java [hunk 1]
```diff
@@ -37,8 +37,8 @@ public class MetadataStatCompositeTest {
     void testInit() {
         MetadataStatComposite statComposite = new MetadataStatComposite();
         Assertions.assertEquals(statComposite.numStats.size(), MetadataEvent.Type.values().length);
-        //(rt)5 * (push,subscribe)2
-        Assertions.assertEquals(5 * 2, statComposite.rtStats.size());
+        //(rt)5 * (push,subscribe,store provider)2
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-metadata/src/test/java/org/apache/dubbo/metrics/metadata/MetadataStatCompositeTest.java:40]: Shouldn't the assertion here be 5*4

File: dubbo-metrics/dubbo-metrics-metadata/src/main/java/org/apache/dubbo/metrics/metadata/event/StoreProviderMetadataListener.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+    @Override
+    public boolean isSupport(MetricsEvent event) {
+        return event instanceof MetadataEvent.StoreProviderMetadataEvent && ((MetadataEvent) event).isAvailable();
+    }
+
+    @Override
+    public void onEvent(MetadataEvent.StoreProviderMetadataEvent event) {
+        event.getCollector().increment(event.getSource().getApplicationName(), MetadataEvent.Type.S_P_TOTAL);
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-metadata/src/main/java/org/apache/dubbo/metrics/metadata/event/StoreProviderMetadataListener.java:35]: Why does the interface level store applicationName?

**Review by wxbty — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — APPROVED** [review]

File: dubbo-metrics/dubbo-metrics-metadata/src/main/java/org/apache/dubbo/metrics/metadata/event/StoreProviderMetadataListener.java (hunk 2 above)
Comment (dengWuuu) [reply dubbo-metrics/dubbo-metrics-metadata/src/main/java/org/apache/dubbo/metrics/metadata/event/StoreProviderMetadataListener.java]: done

File: dubbo-compatible/src/main/java/com/alibaba/dubbo/config/spring/context/annotation/EnableDubbo.java [hunk 3]
```diff
@@ -52,7 +52,7 @@
     String[] scanBasePackages() default {};
 
     /**
-     * Type-safe alternative to {@link #scanBasePackages()} for specifying the packages to
+     * ApplicationType-safe alternative to {@link #scanBasePackages()} for specifying the packages to
```
[the remark this answers is not in the document]
Comment (dengWuuu) [reply dubbo-compatible/src/main/java/com/alibaba/dubbo/config/spring/context/annotation/EnableDubbo.java]: done

**Review by wxbty — APPROVED** [review]

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**